File: iterableapi/src/main/java/com/iterable/iterableapi/ddl/MatchFpResponse.java
package com.iterable.iterableapi.ddl;

import org.json.JSONException;
import org.json.JSONObject;

public class MatchFpResponse {
    public final boolean isMatch;
    public final String destinationUrl;
    public final int campaignId;
    public final int templateId;
    public final String messageId;

    public MatchFpResponse(boolean isMatch, String destinationUrl, int campaignId, int templateId, String messageId) {
        this.isMatch = isMatch;
        this.destinationUrl = destinationUrl;
        this.campaignId = campaignId;
        this.templateId = templateId;
        this.messageId = messageId;
    }

    public static MatchFpResponse fromJSONObject(JSONObject json) throws JSONException {
        return new MatchFpResponse(
                json.getBoolean("isMatch"),
                json.getString("destinationUrl"),
                json.getInt("campaignId"),
                json.getInt("templateId"),
                json.getString("messageId")
        );
    }
}

File: iterableapi/src/main/java/com/iterable/iterableapi/ddl/MatchFpResponse.kt
package com.iterable.iterableapi.ddl

import org.json.JSONException
import org.json.JSONObject

class MatchFpResponse(
    val isMatch: Boolean,
    val destinationUrl: String,
    val campaignId: Int,
    val templateId: Int,
    val messageId: String
) {

    companion object {
        @Throws(JSONException::class)
        fun fromJSONObject(json: JSONObject): MatchFpResponse {
            return MatchFpResponse(
                json.getBoolean("isMatch"),
                json.getString("destinationUrl"),
                json.getInt("campaignId"),
                json.getInt("templateId"),
                json.getString("messageId")
            )
        }
    }
}
